Mathlib/Order/Partition/Finpartition.lean

```diff
@@ -602,4 +677,37 @@ theorem card_filter_atomise_le_two_pow (ht : t ∈ F) :
 
 end Atomise
 
+section Representatives
+
+/-- Choose representatives from each part of a finpartition, collecting them into a finset. -/
+noncomputable def reprs : Finset α :=
+  P.parts.attach.map ⟨fun p => (P.nonempty_of_mem_parts p.2).choose, by
+    rw [Injective]
+    intro ⟨v1, p1⟩ ⟨v2, p2⟩ eq
+    rw [Subtype.mk.injEq]
+    exact P.eq_of_mem_parts p1 p2 (eq ▸ (P.nonempty_of_mem_parts p1).choose_spec)
+      (P.nonempty_of_mem_parts p2).choose_spec⟩
```

Comment:
What's your use case? This looks a bit weird and non-canonical.

Reply:
For this theorem in #9317:
```lean
theorem notAdj_card_parts_le : (notAdjFinpartition hmax).parts.card ≤ r := by
  let fp := notAdjFinpartition hmax
  by_contra! h
  let z := fp.reprs -- `z` is an `r + 1`-clique in `G`
  have ncf : ¬G.CliqueFree z.card := by
    apply IsNClique.not_cliqueFree (s := z)
    constructor
    swap; · rfl
    intro v hv w hw hn
    norm_cast at hv hw
    contrapose! hn
    exact fp.reprs_injective hv hw (fp.eq_of_mem_parts (fp.part_mem _) (fp.part_mem _)
      (Finpartition.mem_part_ofSetoid_iff_rel.mpr hn) (fp.mem_part _))
  rw [Finpartition.card_reprs] at ncf
  exact absurd (CliqueFree.mono (Nat.succ_le_of_lt h) hmax.1) ncf
```

Reply:
Then could you state the existence of such representatives instead of defining arbitrary ones?

Reply:
Done (as `exists_bijOn_part`).

Reply:
Could you then remove the reprs material from the diff?

Reply:
Done.